connectors/crowdstrike/connector/server_access.py

```diff
@@ -59,13 +62,36 @@ def get_hosts(self, last_model_state_id=None):
                     error_detail = host_lookup["body"]["errors"]
                     for err in error_detail:
                         raise DatasourceFailure(err["message"], err["code"])
-                if len(hosts) < host_lookup['body']['meta']['pagination']['total']:
-                    offset = len(hosts)
-                    host_lookup = discover.query_hosts(filter=result_filter, limit=DISCOVER_PAGE_SIZE, offset=offset)
+                hosts = remove_duplicate_hosts(hosts)    
```

Comment:
The problem is the next line.
If we have duplicates at this point in the hosts list then there is a chance we do not get all of the hosts.
For example:
We know the total.
We go through and in the process of ensuring we get all the hosts we have a couple duplicates then when it checks the length of the hosts against the total it may seem like we got them all and will skip trying to get any more.
Let me know if that makes sense.